app/src/main/java/com/example/henzoshimada/feeltrip/MainScreen.java

```diff
@@ -129,7 +129,10 @@ protected void onCreate(Bundle savedInstanceState) {
 
             @Override
             public void onCheckedChanged(CompoundButton arg0, boolean isChecked) {
-                Toast.makeText(getApplicationContext(),"Toggled filter for recent posts!",Toast.LENGTH_SHORT).show();
+                //Toast.makeText(getApplicationContext(),"Toggled filter for recent posts!",Toast.LENGTH_SHORT).show();
+                FeelTripApplication.getFilterController().setPastweekfilter(isChecked);
+                FeelTripApplication.loadFromElasticSearch();
+                FeelTripApplication.getMoodAdapter(getBaseContext()).notifyDataSetChanged();
```

Comment:
Set filter, grab filter query, push returned data to adapter and refresh. All in the click of this toggle switch.